File: CustomToast/app/src/main/java/com/example/customtoast/MainActivity.java
package com.example.customtoast;

import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {
Button btnToast;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        btnToast=findViewById(R.id.btnToast);
//        Default toast
//        btnToast.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                Toast.makeText(getApplicationContext(), "This is my first Toast", Toast.LENGTH_LONG).show();
//            }
//        });



//      Custom Toast



        btnToast.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast toast = new Toast(getApplicationContext());
                View view=getLayoutInflater().inflate(R.layout.custom_toast_layout,(ViewGroup) findViewById(R.id.viewContainer));
                toast.setView(view);

                TextView txtMsg = view.findViewById(R.id.txtMsg);
                txtMsg.setText("Message sent Successfully!");
                toast.setDuration(Toast.LENGTH_LONG);
//                toast.setGravity(Gravity.TOP|Gravity.END,0,0);
                toast.setGravity(Gravity.CENTER,0,0);
                toast.show();
            }
        });


    }
}

File: KotlinBasic/app/src/main/java/com/example/kotlinbasic/MainActivity.kt
package com.example.kotlinbasic

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.kotlinbasic.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding  // Replace with your actual binding class

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.btnAdd.setOnClickListener {
            val no1 = binding.edtNo1.text.toString().toIntOrNull() ?: return@setOnClickListener
            val no2 = binding.edtNo2.text.toString().toIntOrNull() ?: return@setOnClickListener
            val sum = no1 + no2
            Toast.makeText(this@MainActivity, "Sum is $sum", Toast.LENGTH_LONG).show()
            binding.txtSum.text = "Sum is $sum"
        }
    }
}